pkg/cmd/server/api/v1/types.go

```diff
@@ -732,6 +732,9 @@ type RequestHeaderIdentityProvider struct {
 
 	// ClientCA is a file with the trusted signer certs.  If empty, no request verification is done, and any direct request to the OAuth server can impersonate any identity from this provider, merely by setting a request header.
 	ClientCA string `json:"clientCA"`
+	// ClientCommonNames is an optional list of common names to require a match from. If empty, any client certificate validated against the clientCA bundle is considered authoritative.
+	ClientCommonNames []string `json:"clientCommonNames"`
```

Comment:
To me, `clientCommonNames` sounds like aliases for the client, not a list of acceptable clients. Also, a blacklist of common names is not necessarily a bad thing to have. For example, if there are other components of the network that should never be attempting this access, then blacklisting them could potentially thwart an attack.